modules/gateway/conn.go

```diff
@@ -24,13 +24,13 @@ func (pc peerConn) RPCAddr() modules.NetAddress {
 // communication protocol.
 func (g *Gateway) dial(addr modules.NetAddress) (net.Conn, error) {
 	dialer := &net.Dialer{
-		Timeout: dialTimeout,
-		Cancel:  g.threads.StopChan(),
+		Cancel:   g.threads.StopChan(),
+		Deadline: time.Now().Add(connStdDeadline),
+		Timeout:  dialTimeout,
 	}
 	conn, err := dialer.Dial("tcp", string(addr))
 	if err != nil {
 		return nil, err
 	}
-	conn.SetDeadline(time.Now().Add(connStdDeadline))
```

Comment:
I don't think these are equivalent. The `Dial` docstring says:

>// Deadline is the absolute point in time after which dials
        // will fail. If Timeout is set, it may fail earlier.
        // Zero means no deadline, or dependent on the operating system
        // as with the Timeout option.
        Deadline time.Time

So this deadline only applies to the dial operation, not the connection. Weird since `Timeout` seems like the same thing, but that's what it says.